plugins/out_opensearch/opensearch.c

```diff
@@ -336,38 +339,50 @@ static int opensearch_format(struct flb_config *config,
     }
 
     /*
-     * If logstash format and id generation are disabled, pre-generate
+     * If data stream mode is enabled without record accessor syntax
+     * or logstash format and id generation are disabled, pre-generate
      * the index line for all records.
      *
      * The header stored in 'j_index' will be used for the all records on
      * this payload.
      */
-    if (ctx->logstash_format == FLB_FALSE && ctx->generate_id == FLB_FALSE && ctx->ra_index == NULL) {
-        flb_time_get(&tms);
-        gmtime_r(&tms.tm.tv_sec, &tm);
-        strftime(index_formatted, sizeof(index_formatted) - 1,
-                 ctx->index, &tm);
-        index = index_formatted;
-        if (ctx->suppress_type_name) {
-            index_len = flb_sds_snprintf(&j_index,
-                                         flb_sds_alloc(j_index),
-                                         OS_BULK_INDEX_FMT_NO_TYPE,
-                                         ctx->action,
-                                         index);
+
+    enabled_data_stream_no_ra = ctx->data_stream_mode == FLB_TRUE && ctx->ra_data_stream == NULL;
```

Comment:
Please expand these two inlines conditionals

Reply:
Didn't get you, are you suggesting that I do something like:
```
if (enabled_data_stream_no_ra && ctx->generate_id == NULL
    || disabled_logstash_fmt_no_ra && ctx->generate_id == NULL) {
...
```
Let me know what you mean